### PR TITLE
[cdc-cli][cdc-composer] Add 'flink-config' for pipeline yaml

### PR DESCRIPTION
**Pull request by joyCurry30**
Refer to #2854 
Add 'flink-config' for pipeline yaml, which will overwrite the config from 'flink-conf.yaml'. cc @lvyanquan @PatrickRen

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,6 +56,10 @@ sink:
 pipeline:
   name: mysql-sync-doris
   parallelism: 4
+
+flink-config:
+  execution.checkpointing.interval: 30000
+  execution.checkpointing.mode: EXACTLY_ONC
```
Comment (leonardBang) [inline README.md:62]: We need discuss the user facing API in dev group first and then open a PR, we should be careful when introduce new API for end users.

Comment (joyCurry30) [reply README.md:62]: @leonardBang Thank you for review. I have supplemented the relevant issues accordingly #2854.

**Review by lvyanquan — APPROVED** [review]
LGTM.

File: flink-cdc-cli/src/main/java/com/ververica/cdc/cli/parser/YamlPipelineDefinitionParser.java [hunk 2]
```diff
@@ -43,6 +43,7 @@ public class YamlPipelineDefinitionParser implements PipelineDefinitionParser {
     private static final String SINK_KEY = "sink";
     private static final String ROUTE_KEY = "route";
     private static final String PIPELINE_KEY = "pipeline";
+    private static final String FLINK_CONFIG_KEY = "flink-config";
```
Comment (leonardBang) [inline flink-cdc-cli/src/main/java/com/ververica/cdc/cli/parser/YamlPipelineDefinitionParser.java:46]: I'm not tend to introduce another top level concept(section) to users, less is more, could we make user defined flink config as part fo pipeline section?

**Review by leonardBang — COMMENTED** [review]
Thanks @joyCurry30 for the contribution, I left one comment

File: flink-cdc-cli/src/main/java/com/ververica/cdc/cli/parser/YamlPipelineDefinitionParser.java (hunk 2 above)
Comment (joyCurry30) [reply flink-cdc-cli/src/main/java/com/ververica/cdc/cli/parser/YamlPipelineDefinitionParser.java:46]: Sorry for the late reply. Yes, I agreed with you, we should not give users too much concept. But how do we distinguish which configs belong to flink config? Use "flink.config" as the prefix to distinguish?

File: flink-cdc-cli/src/test/resources/definitions/pipeline-definition-full.yaml [hunk 3]
```diff
@@ -53,3 +53,7 @@ pipeline:
   name: source-database-sync-pipe
   parallelism: 4
   enable-schema-evolution: false
+
+flink-config:
```
Comment (czy006) [inline flink-cdc-cli/src/test/resources/definitions/pipeline-definition-full.yaml:57]: flink-config parallelism and cdc-confg parallelism, how is the priority divided here?

Comment (joyCurry30) [reply flink-cdc-cli/src/test/resources/definitions/pipeline-definition-full.yaml:57]: Currently, if parallelism exists both in the Flink config and CDC config, the Flink config will override the CDC config.

Comment (czy006) [reply flink-cdc-cli/src/test/resources/definitions/pipeline-definition-full.yaml:57]: OK

**Pull request closed without merging**